sklearn/ensemble/tests/test_forest.py

```diff
@@ -1359,3 +1361,21 @@ def test_multi_target(name, oob_score):
     # Try to fit and predict.
     clf.fit(X, y)
     clf.predict(X)
+
+
+def test_random_forest_memmap():
+    # check that random forest supports read-only buffer (#13626)
+    X_orig = np.random.RandomState(0).random_sample((10, 2)).astype(np.float32)
+
+    with NamedTemporaryFile() as tmp:
+        tmp.close()  # necessary under windows
+        mmap = np.memmap(tmp.name, dtype='float32', mode='w+', shape=(10, 2))
+        mmap[...] = X_orig[...]
+
+        X_mmap = np.memmap(tmp.name, dtype='float32', mode='r', shape=(10, 2))
+        y = np.zeros(10)
+
+        RandomForestClassifier(n_estimators=2).fit(X_mmap, y)
```

Comment:
I think the test could be under `sklearn/tree/tests/test_tree.py`, and test a `DecisionTreeRegressor` instead. It kinda feels like that's a more natural place for the test since it's actually testing the splitter.

Reply:
I moved the test, and cleaned it since we actually have a helper to test on memmap arrays.